lib/rouge/lexers/sqf.rb

```diff
@@ -0,0 +1,601 @@
+          unitsbelowheight unlinkitem unlockachievement unregistertask
+          updatedrawicon updatemenuitem updateobjecttree
+          useaiopermapobstructiontest useaisteeringcomponent
+          useaudiotimeformoves userinputdisabled vectoradd vectorcos
+          vectorcrossproduct vectordiff vectordir vectordirvisual vectordistance
+          vectordistancesqr vectordotproduct vectorfromto vectormagnitude
+          vectormagnitudesqr vectormodeltoworld vectormodeltoworldvisual
+          vectormultiply vectornormalized vectorup vectorupvisual
+          vectorworldtomodel vectorworldtomodelvisual vehicle
+          vehiclecargoenabled vehiclechat vehicleradio
+          vehiclereceiveremotetargets vehiclereportownposition
+          vehiclereportremotetargets vehicles vehiclevarname velocity
+          velocitymodelspace verifysignature vest vestcontainer vestitems
+          vestmagazines viewdistance visiblecompass visiblegps visiblemap
+          visibleposition visiblepositionasl visiblescoretable visiblewatch
+          waituntil waves waypointattachedobject waypointattachedvehicle
+          waypointattachobject waypointattachvehicle waypointbehaviour
+          waypointcombatmode waypointcompletionradius waypointdescription
+          waypointforcebehaviour waypointformation waypointhouseposition
+          waypointloiterradius waypointloitertype waypointname waypointposition
+          waypoints waypointscript waypointsenableduav waypointshow
+          waypointspeed waypointstatements waypointtimeout
+          waypointtimeoutcurrent waypointtype waypointvisible weaponaccessories
+          weaponaccessoriescargo weaponcargo weapondirection weaponinertia
+          weaponlowered weapons weaponsitems weaponsitemscargo weaponstate
+          weaponsturret weightrtd west wfsidetext while wind winddir windrtd
+          windstr wingsforcesrtd with worldname worldsize worldtomodel
+          worldtomodelvisual worldtoscreen
+        )
+      end
+
+      def self.detect?(text)
+        false
+      end
+
+      state :root do
+        # Whitespace
+        rule %r"\s+", Text
+
+        # Preprocessor instructions
+        rule %r"/\*.*?\*/"m, Comment::Multiline
+        rule %r"//.*\n", Comment::Single
+        rule %r"#(define|undef|if(n)?def|else|endif|include)", Comment::Preproc
+        rule %r"\\\r?\n", Comment::Preproc
+        rule %r"__(EVAL|EXEC|LINE__|FILE__)", Name::Builtin
+
+        # Literals
+        rule %r"\".*?\"", Literal::String
+        rule %r"'.*?'", Literal::String
+        rule %r"(\$|0x)[0-9a-fA-F]+", Literal::Number::Hex
+        rule %r"[0-9]+(\.)?(e[0-9]+)?", Literal::Number::Float
+
+        # Symbols
+        rule %r"[\!\%\&\*\+\-\/\<\=\>\^\|]", Operator
+        rule %r"[\(\)\{\}\[\]\,\:\;]", Punctuation
+
+        # Identifiers (variables and functions)
+        rule %r"[a-zA-Z0-9_]+" do |m|
+          name = m[0].downcase
+          if self.class.wordoperators.include? name
```

Comment:
Use declarative DSL like Ruby's lexer: https://github.com/jneen/rouge/blob/master/lib/rouge/lexers/ruby.rb#L203

Reply:
I actually prefer it this way for keyword lists - in my experience it tends to perform better than large `|` regexes.